aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -75,7 +75,9 @@ SparseTensor new_sparse(const TensorOptions& options) {
 /*** Helper methods ***/
 
 /* Pointer-copy init */
-SparseTensor new_with_tensor_sparse(const LongTensor& indices, const Tensor& values_) {
+SparseTensor new_with_tensor_sparse(const LongTensor& indices_, const Tensor& values__) {
```

Comment:
how should I fix it?

Reply:
I see, but wouldn't it be nice to assume contiguous since ctor?

Reply:
But you can't assume that... for reasons I described above.

None of the sparse op modifies in-place, so this isn't a perf problem as well.

Reply:
If we call `contiguous()`explicitly, then we can assume that? By "assuming", I meant `ctor -> indices and values are contiguous -> op1 -> op2 -> ...`. I am not concerning about perf here.

Reply:
No you can't... people can still `._indices().as_strided_()`.

Reply:
that would be in op1, to have `_indices()`, a sparse tensor is already constructed via ctor

Reply:
I finally see what you mean, since people can always make indices non-contiguous after ctor, hence every time a method takes in a SparseTensor and requires contiguity should always check that explicitly or have a kernel handle it. It is pointless to assure that inside the constructor. Overall user should always think of SparseTensor as a collection of indices and values, within which contiguous and coalesce are not guaranteed.